asapdiscovery-ml/asapdiscovery/ml/trainer.py

```diff
@@ -518,6 +536,13 @@ def initialize(self):
                 logfile=str(self.log_file.name),
             ).getLogger()
 
+        # check S3 settings so that
+        if self.upload_to_s3:
+            try:
+                self.s3_settings = S3Settings()
```

Comment:
check if something  was specified for `self.s3_settings` before overwriting? not sure what all is stored in the `S3Settings` object

Reply:
Good point.

Reply:
Fixed.